src/rdkafka.c

```diff
@@ -1103,22 +1103,23 @@ rd_kafka_t *rd_kafka_new (rd_kafka_type_t type, rd_kafka_conf_t *conf,
 
 	call_once(&rd_kafka_global_init_once, rd_kafka_global_init);
 
-
-	if (!conf)
-		conf = rd_kafka_conf_new();
+        /* Use a copy of conf if supplied, otherwise generate a new one so we can always
+         * delete it upon return without additional conditionals
+         */
+        rd_kafka_conf_t* use_conf = (conf ? rd_kafka_conf_dup(conf) : rd_kafka_conf_new());
```

Comment:
Inline (not at start of scope) variable declarations are not allowed (for portability's sake), put it with the rest of the variables at the top.